superset/views/chart/filters.py

```diff
@@ -14,14 +14,17 @@
 # KIND, either express or implied.  See the License for the
 # specific language governing permissions and limitations
 # under the License.
+from typing import Callable
+
 from sqlalchemy import or_
+from sqlalchemy.orm.query import Query
 
 from superset import security_manager
 from superset.views.base import BaseFilter
 
 
 class SliceFilter(BaseFilter):  # pylint: disable=too-few-public-methods
-    def apply(self, query, value):
+    def apply(self, query: Query, value: str) -> Query:
```

Comment:
@dpgaspar I'm not sure what the type of `value` should be. Looking at FAB I've seen it used as if it were a `str`. Additionally I've seen other causes where `value` is `func` so I'm not sure what's right here.

Reply:
Note this comment outdated and now is typed as `value: Optional[Any]`.